### PR TITLE
Write timelines to separate files

### LINKED ISSUE #74930: Chrome tracing tools not able to read multiple timelines from one file generated by the integration_test package
While migrating the [testing sample](https://github.com/flutter/samples/tree/master/testing_app) under flutter/samples to the `integration_test` package, I am not able to load the `integration_response_data.json` file into `chrome://tracing` without storing the timelines into separate files.
## Steps to Reproduce

1. driver.dart
```
// Copyright 2020 The Flutter team. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:integration_test/integration_test_driver.dart';

Future<void> main() => integrationDriver();
```
2. perf_test.dart
```
// Copyright 2020 The Flutter team. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:integration_test/integration_test.dart';
import 'package:testing_app/main.dart';

void main() {
  group('Testing App Performance Tests', () {
    final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized()
        as IntegrationTestWidgetsFlutterBinding;

    // The fullyLive frame policy simulates
    // the way Flutter responds to animations.
    binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;

    testWidgets('Scrolling test', (tester) async {
      await tester.pumpWidget(TestingApp());

      // Create variables for finders that are used multiple times.
      final listFinder = find.byType(ListView);
      final scroller = tester.widget<ListView>(listFinder).controller;

      // Record the performance timeline of the following operations.
      await binding.traceAction(
        () async {
          // Record the performance summary as the app scrolls through
          // the list of items.
          await binding.watchPerformance(
            () async {
              // Quickly scroll all the way down.
              await scroller.animateTo(
                7000,
                duration: const Duration(seconds: 1),
                curve: Curves.linear,
              );
              await tester.pumpAndSettle();

              // Quickly scroll back up all the way.
              await scroller.animateTo(
                -7000,
                duration: const Duration(seconds: 1),
                curve: Curves.linear,
              );
              await tester.pumpAndSettle();
            },
            // Send the performance summary to the driver.
            reportKey: 'scrolling_summary',
          );
        },
        // Send the timeline data to the driver.
        // This timeline can be opened in the Chrome browser's tracing tools
        // by navigating to chrome://tracing.
        reportKey: 'scrolling_timeline',
      );
    });

    testWidgets('Favorites operations test', (tester) async {
      await tester.pumpWidget(TestingApp());

      // Record the performance timeline of the following operations.
      await binding.traceAction(
        () async {
          // Record the performance summary as operations are performed
          // on the favorites list.
          await binding.watchPerformance(
            () async {
              // Create a list of icon keys.
              final iconKeys = [
                'icon_0',
                'icon_1',
                'icon_2',
              ];

              // Add first three items to favorites.
              for (var icon in iconKeys) {
                // Tap onto the icon.
                await tester.tap(find.byKey(ValueKey(icon)));
                await tester.pumpAndSettle();

                // Verify if appropriate message appears.
                expect(find.text('Added to favorites.'), findsOneWidget);
              }

              // Tap onto the favorites button on the AppBar.
              // The Favorites List should appear.
              await tester.tap(find.text('Favorites'));
              await tester.pumpAndSettle();

              final removeIconKeys = [
                'remove_icon_0',
                'remove_icon_1',
                'remove_icon_2',
              ];

              // Remove all the items from favorites.
              for (final iconKey in removeIconKeys) {
                // Tap onto the remove icon.
                await tester.tap(find.byKey(ValueKey(iconKey)));
                await tester.pumpAndSettle();

                // Verify if appropriate message appears.
                expect(find.text('Removed from favorites.'), findsOneWidget);
              }
            },
            // Send the performance summary to the driver.
            reportKey: 'favorites_operations_summary',
          );
        },
        // Send the timeline data to the driver.
        // This timeline can be opened in the Chrome browser's tracing tools
        // by navigating to chrome://tracing.
        reportKey: 'favorites_operations_timeline',
      );
    });
  });
}
```
3. Run the test using `flutter drive --driver integration_test/driver.dart --target integration_test/perf_test.dart --profile`
4. Load the `integration_response_data.json` file into `chrome://tracing`

**Expected results:** 
The file should load and the data should show up with charts.

**Actual results:** 
The file loads, but no charts or data appear.

If I add a [custom implementation](https://github.com/abd99/samples/blob/integration-test/testing_app/integration_test/driver.dart) to the `responseDataCallback` and store the timelines into different files, it loads successfully.

[Link](https://github.com/flutter/samples/pull/633#discussion_r557562722) to the discussion on the PR.

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-arm, locale
    en-US)
    • Flutter version 1.22.5 at /Users/abd/Developer/flutter
    • Framework revision 7891006299 (7 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/abd/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/abd/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[!] Android Studio (version 4.1)
    • Android Studio at /Users/abd/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (1 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64 • Android 11
      (API 30) (emulator)

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/74930

@johnpryan

### CODE REVIEW

File: packages/integration_test/lib/integration_test_driver.dart [hunk 1]
```diff
@@ -24,23 +24,29 @@ String testOutputsDirectory =
...
-  final File file = fs.file(path.join(
-    destinationDirectory,
-    '$testOutputFilename.json',
-  ));
-  final String resultString = _encodeJson(data, true);
-  await file.writeAsString(resultString);
+  for (final String key in data.keys.where((String key) => key != 'screenshots')) {
+    final File file = fs.file(path.join(
+      destinationDirectory,
+      '${testOutputFilename}_$key.json',
```
Comment (johnpryan) [inline packages/integration_test/lib/integration_test_driver.dart:45]: Just a drive by question, will the file names match the name of the test?

Comment (dnfield) [reply packages/integration_test/lib/integration_test_driver.dart:45]: They'll match whatever was given as the 'reportKey' argument. In the original bug, you'd get `integration_response_data_ favorites_operations_timeline.json` etc.

Comment (dnfield) [reply packages/integration_test/lib/integration_test_driver.dart:45]: (See the test below for sample file names)

**Review by johnpryan — APPROVED** [review]

File: packages/integration_test/lib/integration_test.dart [hunk 2]
```diff
@@ -297,12 +297,17 @@ https://flutter.dev/docs/testing/integration-tests#testing-on-firebase-test-lab
   ///
   /// The `streams` and `retainPriorEvents` parameters are passed as-is to
   /// [traceTimeline].
+  ///
+  /// The `reportKey` must not be `'screenshots'`, which is reserved.
```
Comment (blasten) [inline packages/integration_test/lib/integration_test.dart:301]: nit: reserved for?

**Review by blasten — APPROVED** [review]

File: packages/integration_test/lib/integration_test.dart (hunk 2 above)
Comment (dnfield) [reply packages/integration_test/lib/integration_test.dart:301]: Added a reference to `takeScreenshot`

**Comment by dnfield** [comment]
Something still weird on Windows, adding some prints to see fi I can get more info :\

**Comment by dnfield** [comment]
Ok. `fs.directory.childFile` does not handle paths properly for Windows. Switched to the proper path api.

**Pull request merged**